drivers/timer/npcx_itim_timer.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2021 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_itim_timer
+
+/**
+ * @file
+ * @brief Nuvoton NPCX kernel device driver for  "system clock driver" interface
+ *
+ * This file contains a kernel device driver implemented by the internal
+ * 64/32-bit timers in Nuvoton NPCX series. Via these two kinds of timers, the
+ * driver provides an standard "system clock driver" interface.
+ *
+ * It includes:
+ * - A system timer based on an ITIM64 (Internal 64-bit timer) instance, clocked
+ *   by APB2 which freq is the same as CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC.
+ * - Provide a 64-bit cycles reading and ticks computation based on it.
+ * - Its prescaler is set to 1 and provide the kernel cycles reading without
+ *   handling overflow mechanism.
+ * - After ec entered "sleep/deep sleep" power state which is used for better
+ *   power consumption, then its clock will stop.
+ *
+ * - A event timer based on an ITIM32 (Internal 32-bit timer) instance, clocked
+ *   by LFCLK which frequency is 32KHz and still activated when ec entered
+ *   "sleep/deep sleep" power state.
+ * - Setup time-out event and handle z_clock_announce() in its ISR.
+ * - Its prescaler is set to 1 and provide timeout event mechanism.
+ * - Compensate system timer which clock is gating after ec left
+ *   "sleep/deep sleep" power state.
```

Comment:
What are these last three bullet points about?

Reply:
Indeed, it looks ambiguous. I will modify them for better explanations.

Reply:
Done.